docs/designs/keras_callback.md

```diff
@@ -0,0 +1,196 @@
+# Design to Support Keras Callback
+
+This document describes the design for supporting callback to customize the behavior of model in ElasticDL.
```

Comment:
Not customize the model, but the model training process.

Reply:
Not only the training process but also the prediction and evaluation process.
"Customize the behavior during the model iteration?"